src/pages/DistributePage/DistributePage.tsx

```diff
@@ -48,29 +48,30 @@ function DistributePage() {
   const [updateAmount, setUpdateAmount] = useState(0);
   const [selectedVaultId, setSelectedVaultId] = useState('');
 
-  const contracts = useContracts();
   const currentOrg = useCurrentOrg();
+  const submitDistribution = useSubmitDistribution();
+  const currentUser = useCurrentUserForEpoch(Number(epochId));
+
   const { isLoading: vaultLoading, data: vaults } = useVaults(
     currentOrg.data?.id as number
   );
 
-  const { uploadEpochRoot } = useDistributor();
-  const selectedVault = vaults?.find(v => v.id === Number(selectedVaultId));
-  const { apeError } = useApeSnackbar();
-  const { mutateAsync } = useSaveEpochDistribution();
-  const { mutateAsync: updateDistributionMutateAsync } =
-    useUpdateDistribution();
-
   const {
     isLoading: isAllocationsLoading,
     isError,
     data,
   } = useGetAllocations(Number(epochId));
 
-  const currentUser = useCurrentUserForEpoch(Number(epochId));
+  const {
+    data: previousDistribution,
+    isLoading: loadingPreviousDistributions,
+  } = usePreviousDistributions(Number(data?.epochs_by_pk?.circle?.id));
```

Comment:
Wrapping the data result in Number casts the potentially undefined type to be a number. `usePreviousDistributions` should be updated to handle undefined type instead of wrapping this in Number